### PR TITLE
Yuqiang/fuzz verifier

### PR DESCRIPTION
**Pull request by smallst**
fix ENG-147
example testing url, check the results in console for now
`http://localhost:3000/editor/verifier/true-names?check_prop=true&languages=javascript`

### todo
- [x] only checking levels without limitation
- [x] add maxSpeed
- [x] add attackDamage
- [x] add upper bound checking
- [ ] use UI to select testing langs (for sinlge levels)
- [ ] use UI to display results
- [ ] use UI to determine if we want to enable binary search
- [x] clean code

### CODE REVIEW

File: app/views/editor/verifier/VerifierTest.js [hunk 1]
```diff
@@ -1,31 +1,19 @@
-// TODO: This file was created by bulk-decaffeinate.
-// Sanity-check the conversion and remove this comment.
-/*
- * decaffeinate suggestions:
- * DS002: Fix invalid constructor
- * DS101: Remove unnecessary use of Array.from
- * DS102: Remove unnecessary code created because of implicit returns
- * DS103: Rewrite code to no longer use __guard__, or convert again using --optional-chaining
```
Comment (nwinter) [inline app/views/editor/verifier/VerifierTest.js:8]: We shouldn't get rid of the decaffeinate comments for stuff we haven't yet fixed (like the `guard` thing)

File: app/views/editor/verifier/VerifierTest.js [hunk 2]
```diff
@@ -115,13 +114,33 @@ module.exports = (VerifierTest = class VerifierTest extends CocoClass {
...
   setupGod () {
-    this.god.setLevel(this.level.serialize({ supermodel: this.supermodel, session: this.session, otherSession: null, headless: true, sessionless: false }))
+    const level = this.level.serialize({ supermodel: this.supermodel, session: this.session, otherSession: null, headless: true, sessionless: false })
+    const hero = this.world.getThangByID('Hero Placeholder')
+    if (this.checkClampedProperties) {
+      level.constrainHeroHealth = true
+      if (!this.clampedProperties.inited) {
+        this.clampedProperties = {
+          inited: true,
+          maxHealth: { current: 1, lower: 1, upper: hero.maxHealth },
```
Comment (nwinter) [inline app/views/editor/verifier/VerifierTest.js:124]: Minimum hero health is something like 6, we can probably start there.

File: app/views/editor/verifier/VerifierTest.js [hunk 3]
```diff
@@ -115,13 +114,33 @@ module.exports = (VerifierTest = class VerifierTest extends CocoClass {
...
+          maxSpeed: { current: 1, lower: 1, upper: hero.maxSpeed },
```
Comment (nwinter) [inline app/views/editor/verifier/VerifierTest.js:125]: Slowest possible speed is something like 3. We also need to check to see if the hero can have too much speed to beat the level, so upper could be like 25.

File: app/views/editor/verifier/VerifierTest.js [hunk 4]
```diff
@@ -115,13 +114,33 @@ module.exports = (VerifierTest = class VerifierTest extends CocoClass {
...
+          attackDamage: { current: 1, lower: 1, upper: hero.attackDamage }
```
Comment (nwinter) [inline app/views/editor/verifier/VerifierTest.js:126]: Attack damage too high could also be a problem. We can start with lower as 4 and upper as 400 or something like this.

File: app/views/editor/verifier/VerifierTest.js (hunk 3 above)
Comment (smallst) [reply app/views/editor/verifier/VerifierTest.js:125]: ya, i would add upper bound checking later. i just want to first made a good logic to check lowest one by one, then copy that for upper bound

File: app/views/editor/verifier/VerifierTest.js (hunk 1 above)
Comment (smallst) [reply app/views/editor/verifier/VerifierTest.js:8]: i'll fix this file

**Review by nwinter — CHANGES_REQUESTED** [review]
Test on greed-traps and some other levels that we know have current solution issues to try more cases.

We shouldn't output suggested min/max property amounts if the upper and lower bounds work, we can just leave them off and don't need to set them.

Didn't have time to test thoroughly.

**Comment by smallst** [comment]
my account can't play greed-traps. this pr is open too long. let's merge it first. if we need to use it and finding bugs, we can fix later

**Pull request merged**